dbt_invoke/properties.py

```diff
@@ -268,104 +278,158 @@ def migrate(
+        state=state,
+        log_level=log_level,
+    )
+    # Parse the manifest file.
+    nodes = _read_manifest(ctx.get('target_path'))['nodes']
+    # Create a placeholder map object
+    # Structure of a migration map
+    # config_path = A string that represents the location of a yaml file
+    #   resource_list = A list of all nodes which were found in this config location
+    #       resource_name = The name which should be used within the yaml file
+    #       resource_desitination = A Path object which describes where the resource lives if we adhere to the 1:1 model -> yaml
+    #       resource_properties = The definition extracted from the config_path for this resource
+    #       resource_index = An integer representing the position within the source yaml which is an OrderedDict object where the properties were found
+    # {
+    #   config_path: {
+    #       resource_list: [
+    #           {
+    #               resource_name: str,
+    #               resource_type: str,
+    #               resource_type_plural: str,
+    #               resource_destination: Path,
+    #               resource_definition: OrderedDict
+    #               resource_index: int
+    #           }
+    #       ]
+    #   }
+    # }
+    migration_map = dict()
+    # Convert dbt ls data into something easier to search
+    resources_map = {resource_dict['name']: resource_dict for resource_dict in transformed_ls_results.values()}
+    # Using the nodes from the manifest create a data structure that keeps track of what existing yaml files we have
+    # and what resources are defined there in each.
+    for node in nodes:
+        # If there is no config for this node move on
+        if not nodes[node]['patch_path']:
+            continue
+        # Parse the patch path to make a full file system path
+        patch_path = ctx.config['project_path'] + '/' + nodes[node]['patch_path'].split('//')[1]
         # Add to-be-created property file paths to migration_map
-        for resource_location, resource_dict in transformed_ls_results.items():
-            migration_map[resource_dict['name']]['path'] = Path(
-                ctx.config['project_path'],
-                resource_location,
-            ).with_suffix('.yml')
-
-        # Try to create each new property file. For each success, save
-        # the index of the resource to remove from the migration
-        # property file.
-        indices_to_remove = defaultdict(list)
-        for resource_name, data in migration_map.items():
-            try:
-                property_file_dict = _get_property_header(
-                    resource_name,
-                    data['resource_type'],
-                    data['properties'],
-                )
-                _utils.write_yaml(data['path'], property_file_dict, mode='x')
-                indices_to_remove[data['resource_type_plural']].append(
-                    data['index']
+        if patch_path not in migration_map:
```

Comment:
I may be missing something obvious, but I am unable to think of a situation where `patch_path` would already be in `migration_map`. Did a situation like that arise during development or testing?

Reply:
The `patch_path` is the current config file where the resource exists so if there is more than one resource defined in that file then as we are building the `migration_map` we will see the value more than once.